# **Análisis Exploratorio De Datos**

In [1]:
import pandas as pd

In [3]:
# DataSet Número 2
ruta = r"C:\Users\Luis Sazo\Desktop\Ventas Supermercados\Data\annex2.csv"
DataSet2 = pd.read_csv(ruta)
DataSet2

Date          Time        Item Code  Quantity Sold (kilo)  \
0       2020-07-01  09:15:07.924  102900005117056                 0.396   
1       2020-07-01  09:17:27.295  102900005115960                 0.849   
2       2020-07-01  09:17:33.905  102900005117056                 0.409   
3       2020-07-01  09:19:45.450  102900005115823                 0.421   
4       2020-07-01  09:20:23.686  102900005115908                 0.539   
...            ...           ...              ...                   ...   
878498  2023-06-30  21:35:13.264  102900005115250                 0.284   
878499  2023-06-30  21:35:14.358  102900011022764                 0.669   
878500  2023-06-30  21:35:20.264  102900005115250                 0.125   
878501  2023-06-30  21:35:21.509  102900011016701                 0.252   
878502  2023-06-30  21:40:48.248  102900011022764                 0.803   

        Unit Selling Price (RMB/kg) Sale or Return Discount (Yes/No)  
0                               7.6           sale                No  
1                               3.2           sale                No  
2                               7.6           sale                No  
3                              10.0           sale                No  
4                               8.0           sale                No  
...                             ...            ...               ...  
878498                         24.0           sale                No  
878499                         12.0           sale                No  
878500                         24.0           sale                No  
878501                          5.2           sale                No  
878502                         12.0           sale                No  

[878503 rows x 7 columns]

In [ ]:
DataSet2.info()
DataSet2.isnull().sum()
# El DataSet no presenta nulos

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 878503 entries, 0 to 878502
Data columns (total 7 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   Date                         878503 non-null  object 
 1   Time                         878503 non-null  object 
 2   Item Code                    878503 non-null  int64  
 3   Quantity Sold (kilo)         878503 non-null  float64
 4   Unit Selling Price (RMB/kg)  878503 non-null  float64
 5   Sale or Return               878503 non-null  object 
 6   Discount (Yes/No)            878503 non-null  object 
dtypes: float64(2), int64(1), object(4)
memory usage: 46.9+ MB


Date                           0
Time                           0
Item Code                      0
Quantity Sold (kilo)           0
Unit Selling Price (RMB/kg)    0
Sale or Return                 0
Discount (Yes/No)              0
dtype: int64

In [16]:
# Promedio de "Unit Selling Price"
PromedioVentaPorUnidad = DataSet2["Unit Selling Price (RMB/kg)"].mean()
print(f"El promedio de venta por unidad es ${round(PromedioVentaPorUnidad, 2)}")


El promedio de venta por unidad es $8.92


In [18]:
Ventas = DataSet2["Sale or Return"].value_counts().get("sale", 0)
Devoluciones = DataSet2["Sale or Return"].value_counts().get("return", 0)

print(f"Total de Ventas: {Ventas}")
print(f"Total de devoluciones: {Devoluciones}")

Total de Ventas: 878042
Total de devoluciones: 461


In [19]:
DataSet2["Discount (Yes/No)"].value_counts()

Discount (Yes/No)
No     831137
Yes     47366
Name: count, dtype: int64

A 831,137 productos se le aplicó un descuento
mientras que a un total de 47,366 productos no

In [20]:
# Guardar DataSet con otro nombre
DataSet2.to_csv("../Data_Limpia/DetalleProductos.csv", index=False)

In [21]:
# Importación de módulo y crear conexión
from sqlalchemy import create_engine

conexion = create_engine (
    "postgresql+psycopg2://postgres:Adelayda1972@localhost:5432/Ventas-Supermercado"
)

In [23]:
# Convertir DataSet en una tabla en PostgreSQL
DataSet2.to_sql(
    "DetalleVentas",
    con=conexion,
    schema="DataSupermercado",
    if_exists="replace",
    index=False
)

503